### ISSUE TITLE
Issues using this with a self-hosted agent pool

### ISSUE THREAD

**Issue by dwgray** [fakhrulhilal/TFSGitDownloader#2]
I've configured this task and it works great for a hosted agent, but my use case is for on premise agent pools.  Do you expect this plugin to work for that?

I am using the case where I specify the full path to the git repo and enable "Allow Scripts to access OAuth token" on the job.

The behavior is that git appears to hang after emitting "Cloning into 'c:\git\SharedTools...'
The directory is created and, but no files are added.

I'm happy to provide any other diagnostics information, just let me know how to collect it.  Thanks in advance for any help.
[gitdownloader.log](https://github.com/fakhrulhilal/TFSGitDownloader/files/2447228/gitdownloader.log)

**Comment by fakhrulhilal** [comment]
Yes, it’s supposed to be working for onpremise too. I use it on my company’s server. I see the log, the build is cancelled. Can you retry again without cancelling the task? Please tell us the agent version you’re using too.

**Comment by fakhrulhilal** [comment]
Hi again,
I think I know the problem. You said that the build is in your on premise server. But you want to download https://mycompany.visualstudio.com/Shared/_git/Tools, I believe it’s not on premise. When downloading repo in different server, the plugin expects the repo to be public, means downloadable without authentication. The best way to test it is by cloning using git console. Can you make it sure?
Currently, this plugin doesn’t support downloading external repo with authentication. It behaves just like what you said, like hanging up without any error message.
